src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CollectionBackedStoreProvider.cs

```diff
@@ -4,167 +4,74 @@
 
 using System;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
 using System.Diagnostics;
-using System.Security.Cryptography;
 using System.Security.Cryptography.X509Certificates;
-using Microsoft.Win32.SafeHandles;
+using System.Threading;
 
 namespace Internal.Cryptography.Pal
 {
     internal class CollectionBackedStoreProvider : IStorePal
     {
-        private readonly X509Certificate2[] _certs;
-        private static readonly SafeEvpPKeyHandle InvalidPKeyHandle = new SafeEvpPKeyHandle(IntPtr.Zero, false);
+        private readonly bool _cloneOnRead;
```

Comment:
Yeah, I was considering splitting the IStorePal into Loader/Exporter/SystemStore.

Windows PAL always copies, because it asks CAPI for a PCERT_CONTEXT then wraps that with a new managed object.  Everything in the Unix PAL gets complicated because OpenSSL (P)X509 objects don't have private key associations, so we manage that all in managed code.  This particularly applies to PFX loading; which is where the cloneOnRead is false.

So really it's:
* Export source: Given a valid PAL object, needs to call an underlying API.  Do not Dispose, do not clone.
* Import source: Emit a certificate by whatever means necessary. In Unix emulation of CAPI PCERT_CONTEXT this means passing hydrated X509Certificate2 objects.  Caller will only read once.  Do not Dispose (arguably "dispose unless the caller asked for it already"). do not clone.
* System Store: To match what Windows does, every call to the Certificates property (which uses CopyTo in the PAL) has to return a new X509Certificate2 object.  Clone is required to match existing lifetime semantics.  (Dispose is skipped because we don't support writing to the LM stores; User Stores read hot from the filesystem every time, different interface implementation).

If you agree that there are three different things here I can split the interface to make each IThing have more apparent semantics (my hesitation was "well... I don't really need to bother the Windows half with this...")